src/briefcase/integrations/subprocess.py

```diff
@@ -214,7 +255,8 @@ def verify(cls, tools):
         tools.subprocess = Subprocess(tools)
         return tools.subprocess
 
-    def run(self, args, stream_output=False, **kwargs):
+    @pause_console_control_if
+    def run(self, args, /, stream_output=False, **kwargs):
```

Comment:
Is there any particular reason for introducing the positional-only marker here? We're trying to proxy the API from `subprocess`, and `run()` et al don't have this requirement, so there will be an inconsistency in the ways these methods can be invoked. This probably won't matter in practice, as anything after the `args` argument will usually be a kwarg - but it's an added complexity in the codebase that doesn't really improve anything.

Reply:
I added the positional-only marker so I could simplify inferring where the `cmdline` is in the call parameters of `run()` and `check_output()`. Without this, devs can technically call `run(args=['ls'])` and it works.

Furthermore, though, both `run()` and `check_output()` appear to have always deviated from the the `subprocess` module.

In the `subprocess` module, these functions support an arbitrary number of positional args that are passed to `Popen`; if one isn't careful, these pos args could easily collide with those in `kwargs`.

For the wrapper `Subprocess.check_output()`, it only takes 1 positional argument (or a kwarg called `args`) that must be a an iterable of stringlikes to execute and additional pos args are not supported at all.

The `Subprocess.run()` wrapper has the same issue but goes further because we added the `stream_output` param.

So, while I agree these are proxies for `subprocess`, I don't think they ever actually had compatible signatures. Ultimately, I felt this positional-only marker mostly enforced an already used convention within `briefcase` and removed a little complexity from this decorator.

Alternatively, we might could take a step back and try to better align these wrappers back with `subprocess`.

Reply:
The divergence from `subprocess` is a fair point; I guess it's more "aspirational" than a firm requirement.

However, unless I'm missing something, the `args` problem you describe doesn't exist AFAIK:
```
In [1]: def foo(args, **kwargs):
   ...:     print(f"{args=}, {kwargs=}")
   ...: 

In [2]: foo('a', b=3)
args='a', kwargs={'b': 3}

In [3]: foo(args='a', b=3)
args='a', kwargs={'b': 3}

In [4]: foo(b=3, args='a')
args='a', kwargs={'b': 3}
```

If you declare a positional argument, that key won't ever appear in the kwargs, no matter how it's passed in.

Reply:
That's true; it just changes the assumptions at bit. Instead of assuming the `cmdline` iterable is the first param of the decorated method, this will assume it always has the name `args`. But that isn't necessarily an inherent problem here.

I can switch to that.